src/nkr_proxy/proxy_app.py

```diff
@@ -198,15 +209,29 @@ def search_index(user_restriction_level, entitlements, search_query):
 
     for n_retry, index_host in enumerate(settings.INDEX_HOSTS):
 
-        full_index_url = '%s/%s/%s' % (index_host, settings.INDEX_NAME, search_query)
-
-        logger.debug(full_index_url)
-
         try:
-            response = http_request(
-                full_index_url,
-                auth=(settings.INDEX_USERNAME, settings.INDEX_PASSWORD)
-            )
+            if method == 'get':
+                full_index_url = '%s/%s/%s' % (index_host, settings.INDEX_NAME, search_query)
+                response = http_request(
+                    full_index_url,
+                    method=method,
+                    auth=(settings.INDEX_USERNAME, settings.INDEX_PASSWORD)
+                )
+                logger.debug('Url: %s' % full_index_url)
+                logger.debug('Search query: %s' % search_query)
+            if method == 'post':
+                full_index_url = '%s/%s/select' % (index_host, settings.INDEX_NAME)
+                post_search_query = search_query.lstrip('select?')
+                headers = {'Accept-Encoding': 'gzip, deflate', 'Content-Type': 'application/x-www-form-urlencoded'}
+                response = http_request(
+                    full_index_url,
+                    method=method,
+                    data=post_search_query,
+                    headers=headers,
+                    auth=(settings.INDEX_USERNAME, settings.INDEX_PASSWORD)
+                )
+                logger.debug('Url: %s' % full_index_url)     
+                logger.debug('Search query: %s' % post_search_query)
```

Comment:
These debug logs could be moved before request. Otherwise if the request fails we don't get these debug lines. For both, get and post.